spec/services/custom_fields/hierarchy/hierarchical_item_service_spec.rb

```diff
@@ -152,6 +152,30 @@
           expect(result).to be_success
           expect(root.reload.position_cache).to eq(27)
         end
+
+        context "if the item or its descendants were assigned as a custom value" do
+          let(:wp_type) { create(:type_task) }
+          let(:project) { create(:project, types: [wp_type]) }
+          let(:work_package) { create(:work_package, project:, type: wp_type) }
+          let(:work_package_with_another_value) { create(:work_package, project:, type: wp_type) }
```

Comment:
:yellow_circle: Naming suggestion:

```suggestion
          let(:work_package_with_child_value) { create(:work_package, project:, type: wp_type) }
```

This makes it much more clear, that the other value is not a random _other_ value, but rather that it's a child value of the deleted one and thus there is a reasonable expectation for it to be deleted as well.

Reply:
done, but will split the test cases still